Classes/Handler/ErrorHandler.php

```diff
@@ -22,153 +36,117 @@ class ErrorHandler
     protected $environment;
 
     /**
-     * @var \Raven_Client
+     * @param array $settings
      */
-    protected $client;
+    public function injectSettings(array $settings)
+    {
+        $this->dsn = $settings['dsn'] ?? '';
+        $this->environment = $settings['environment'] ?? '';
+    }
 
     /**
      * Initialize the raven client and fatal error handler (shutdown function)
      */
     public function initializeObject()
     {
-        $client = new \Raven_Client($this->dsn);
-        $errorHandler = new \Raven_ErrorHandler($client, true);
-        $errorHandler->registerShutdownFunction();
-        $this->client = $client;
+        \Sentry\init(['dsn' => $this->dsn]);
+
+        $options = Hub::getCurrent()->getClient()->getOptions();
+        $options->setEnvironment($this->environment);
+        $options->setRelease($this->getReleaseFromReleaseFile());
+        $options->setProjectRoot(FLOW_PATH_ROOT);
+        $options->setPrefixes([FLOW_PATH_ROOT]);
 
         $this->setTags();
-        $this->setEnvironment();
-        $this->setRelease();
-        $this->setAppPath();
-        $this->setPrefixes();
 
-        $this->emitSentryClientCreated($client);
+        $this->emitSentryClientCreated();
     }
 
     /**
-     * Explicitly handle an exception, should be called from an exception handler (in Flow or TypoScript)
+     * Explicitly handle an exception, should be called from an exception handler (in Flow or Fusion)
      *
      * @param object $exception The exception to capture
      * @param array $extraData Additional data passed to the Sentry sample
      */
-    public function handleException($exception, array $extraData = array())
+    public function handleException($exception, array $extraData = []): void
     {
-        if (!$this->client instanceof \Raven_Client) {
-            return;
-        }
 
         if (!$exception instanceof \Throwable) {
             return;
         }
 
-        $this->setUser();
-
-        $tags = array('code' => $exception->getCode());
-        if ($exception instanceof \Neos\Flow\Exception) {
+        if ($exception instanceof WithReferenceCodeInterface) {
             $extraData['referenceCode'] = $exception->getReferenceCode();
         }
 
-        $this->client->captureException($exception, array(
-                'message' => $exception->getMessage(),
-                'extra' => $extraData,
-                'tags' => $tags
-            )
-        );
+        Hub::getCurrent()->configureScope(function (Scope $scope) use ($exception, $extraData): void {
+            $scope->setUser(['username' => $this->getCurrentUserName()]);
+            $scope->setTag('code', (string)$exception->getCode());
+
+            foreach ($extraData as $extraDataKey => $extraDataValue) {
+                $scope->setExtra($extraDataKey, $extraDataValue);
+            }
+        });
+
+        \Sentry\captureException($exception);
     }
 
     /**
      * Set tags on the raven context
```

Comment:
Nit-pick: I believe they're no longer calling it Raven.

```suggestion
      * Set tags on the Sentry client context
```